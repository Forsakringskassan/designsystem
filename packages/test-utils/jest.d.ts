export * from "./dist/types/jest";

declare global {
    namespace jest {
        interface Matchers<R> {
            /**
             * Expects element to have focus.
             */
            toHaveFocus(): R;
        }
    }
}
