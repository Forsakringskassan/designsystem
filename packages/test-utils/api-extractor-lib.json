{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/lib.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/lib.d.ts"
    }
}
