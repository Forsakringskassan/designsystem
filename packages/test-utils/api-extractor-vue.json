{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/vue/index.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/vue.d.ts"
    }
}
