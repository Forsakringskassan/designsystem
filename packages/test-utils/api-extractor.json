{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "tsdocMetadata": {
        "tsdocMetadataFilePath": "<projectFolder>/dist/types/tsdoc-metadata.json"
    },
    "dtsRollup": {
        "enabled": false
    }
}
