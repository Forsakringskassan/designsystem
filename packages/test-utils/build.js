const fs = require("node:fs/promises");
const path = require("node:path");
const esbuild = require("esbuild");

const type = {
    cjs: "commonjs",
    esm: "module",
};

async function build() {
    for (const format of ["cjs", "esm"]) {
        const outdir = `dist/${format}`;
        const common = {
            logLevel: "info",
            sourcemap: true,
            external: ["vue", "@fkui/*"],
            bundle: true,
            platform: "node",
            format,
            target: "node20",
            outdir,
        };
        await esbuild.build({
            ...common,
            entryPoints: ["src/lib.ts", "src/jest.ts"],
        });
        await esbuild.build({
            ...common,
            entryPoints: [{ in: "src/vue/index.ts", out: "vue" }],
            define: {
                "process.env.NODE_ENV": JSON.stringify("development"),
            },
        });
        const pkg = JSON.stringify({ type: type[format] }, null, 2);
        await fs.writeFile(path.join(outdir, "package.json"), pkg, "utf-8");
    }
}

build().catch((err) => {
    console.error(err);
});
