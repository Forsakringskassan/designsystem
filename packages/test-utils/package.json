{
  "name": "@fkui/test-utils",
  "version": "6.6.0",
  "description": "FKUI test utils for Jest",
  "keywords": [
    "fkui",
    "designsystem",
    "test",
    "jest",
    "vue",
    "cypress"
  ],
  "homepage": "https://designsystem.forsakringskassan.se/",
  "bugs": "https://github.com/Forsakringskassan/designsystem/issues",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Forsakringskassan/designsystem.git",
    "directory": "packages/test-utils"
  },
  "license": "MIT",
  "author": "Försäkringskassan",
  "sideEffects": [
    "./dist/cjs/jest.js",
    "./dist/esm/jest.js",
    "./src/matchers/augmentations.ts",
    "./src/matchers/index.ts"
  ],
  "exports": {
    ".": {
      "types": "./dist/types/lib.d.ts",
      "import": "./dist/esm/lib.js",
      "require": "./dist/cjs/lib.js"
    },
    "./jest": {
      "types": "./dist/types/jest.d.ts",
      "import": "./dist/esm/jest.js",
      "require": "./dist/cjs/jest.js"
    },
    "./vue": {
      "types": "./dist/types/vue.d.ts",
      "import": "./dist/esm/vue.js",
      "require": "./dist/cjs/vue.js"
    }
  },
  "main": "dist/cjs/lib.js",
  "types": "dist/types/lib.d.ts",
  "files": [
    "dist",
    "jest.d.ts",
    "vue.d.ts"
  ],
  "scripts": {
    "prebuild": "rimraf dist temp && tsc",
    "build": "node build.js",
    "postbuild": "fk-api-extractor --patch-augmentations api-extractor.json api-extractor-*.json",
    "clean": "rimraf .jest-cache dist temp test-results",
    "prepack": "release-prepack --bundle --retain-scripts",
    "postpack": "release-postpack",
    "prepublishOnly": "release-prepublish --bundle --retain-scripts",
    "test": "jest"
  },
  "jest": {
    "preset": "@forsakringskassan/jest-config-vue"
  },
  "devDependencies": {
    "@vue/server-renderer": "3.5.13",
    "@vue/test-utils": "2.4.6",
    "esbuild": "0.25.3",
    "jest-environment-jsdom": "29.7.0",
    "vue": "3.5.13"
  },
  "peerDependencies": {
    "jest": "^26.0.0 || ^27.0.0 || ^28.0.0 || ^29.0.0",
    "vue": "^3.0.0"
  },
  "peerDependenciesMeta": {
    "vue": {
      "optional": true
    }
  },
  "engines": {
    "node": ">= 20",
    "npm": ">= 7"
  }
}
