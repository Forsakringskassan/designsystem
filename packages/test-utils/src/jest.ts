/* imported for side-effects only */
import "./matchers";

export * from "./lib";
