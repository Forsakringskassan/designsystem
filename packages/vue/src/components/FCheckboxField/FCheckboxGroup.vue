<template>
    <f-fieldset :id="id" :name="name" v-bind="$attrs">
        <template #label>
            <!-- @slot Slot for label content. This slot is required. -->
            <slot name="label"></slot>
        </template>

        <template v-if="$slots.tooltip" #tooltip>
            <!-- @slot Slot for tooltip. -->
            <slot name="tooltip"></slot>
        </template>

        <template #description="{ descriptionClass, discreteDescriptionClass }">
            <!--
                @slot Optional slot for description. See [FLabel](#/Components/FLabel) for details.
                @binding {string[]} descriptionClass CSS classes for primary description content.
                @binding {string[]} discreteDescriptionClass CSS classes for format description.
            -->
            <slot name="description" v-bind="{ descriptionClass, discreteDescriptionClass }"></slot>
        </template>

        <template #error-message>
            <!-- @slot Slot for displaying single or several error messages. -->
            <slot name="error-message"></slot>
        </template>

        <!-- @slot Slot for main content, usually checkboxes. -->
        <slot name="default"></slot>
    </f-fieldset>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { ElementIdService } from "@fkui/logic";
import { FFieldset } from "../FFieldset";
import { TranslationMixin } from "../../plugins";

/**
 * @deprecated `<f-checkbox-group>` is deprecated.
 * See migration guide: https://forsakringskassan.github.io/latest/guide/migration/migrating-to-fieldset.html
 */
export default defineComponent({
    name: "FCheckboxGroup",
    components: { FFieldset },
    mixins: [TranslationMixin],
    inheritAttrs: false,
    props: {
        /**
         * The id for the fieldset id attribute.
         * If the prop is not set the id will be generated.
         */
        id: {
            type: String,
            required: false,
            default: () => ElementIdService.generateElementId(),
        },
        /**
         * The name of the checkbox group.
         */
        name: {
            type: String,
            required: true,
        },
    },
});
</script>
