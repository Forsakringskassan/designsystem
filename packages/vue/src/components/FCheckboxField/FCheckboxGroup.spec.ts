import "html-validate/jest";
import { VueWrapper, mount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import FCheckboxGroup from "./FCheckboxGroup.vue";

function createWrapper({
    props = {},
    slots = {},
    attrs = {},
} = {}): VueWrapper {
    return mount(FCheckboxGroup, {
        attrs: { ...attrs },
        props: { id: "someId", name: "someName", ...props },
        slots: { ...slots },
    });
}

describe("snapshots", () => {
    it("should match snapshot with default classes and filled slots", async () => {
        const wrapper = mount({
            components: { FCheckboxGroup },
            template: /* HTML */ `
                <f-checkbox-group name="someName">
                    <template #label> Label </template>
                    <template #tooltip> Tooltip </template>
                    <template #description> Description </template>
                    <template #error-message> Error message </template>
                    <input type="checkbox" />
                    <input type="checkbox" />
                </f-checkbox-group>
            `,
        });

        await flushPromises();
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe("props", () => {
    it("should generate id when unset", () => {
        const wrapper = createWrapper({ props: { id: undefined } });
        expect(wrapper.element.id).toBe("fkui-vue-element-0001");
    });
});

describe("attributes", () => {
    it("should pass attributes", () => {
        const wrapper = createWrapper({
            attrs: {
                disabled: true,
                required: true,
            },
        });
        const input = wrapper.get("fieldset");

        expect(input.attributes("disabled")).toBeDefined();
        expect(input.attributes("required")).toBeDefined();
    });
});

describe("events", () => {
    it("should pass listeners", async () => {
        const foobar = jest.fn();
        const wrapper = createWrapper({
            attrs: { onFoobar: foobar },
        });
        const element = wrapper.get("fieldset");
        await element.trigger("foobar");
        expect(foobar).toHaveBeenCalled();
    });
});
