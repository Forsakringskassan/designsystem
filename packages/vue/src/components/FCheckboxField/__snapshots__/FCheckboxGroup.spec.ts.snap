// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot with default classes and filled slots 1`] = `
<fieldset
  class="fieldset checkbox-group"
  id="fkui-vue-element-0001"
>
  <legend
    class="label sr-only"
  >
    
    
     Label 
    
    
    <span
      class="sr-only"
    >
      <span>
        Grupp med 2 kryssrutor
      </span>
    </span>
    
    
     Description 
    
    
    
    
     Error message 
    
    
  </legend>
  <span
    aria-live="polite"
    class="sr-only"
    data-test="checked-boxes"
  >
    <span>
      0 kryssad av 2
    </span>
  </span>
  
  <div
    class="sr-separator"
  >
    <div
      aria-hidden="true"
      class="tooltip-before"
    >
      <div
        class="label tooltip-before__label"
      >
        
        
         Label 
        
        
      </div>
    </div>
    
    
     Tooltip 
    
    
  </div>
  <div
    aria-hidden="true"
    class="label fieldset__label"
  >
    
    
     Description 
    
    
    
    
     Error message 
    
    
  </div>
  
  <div
    class="fieldset__content checkbox-group__content"
  >
    
    
    <input
      type="checkbox"
    />
    <input
      type="checkbox"
    />
    
    
  </div>
</fieldset>
`;
